dev/breeze/src/airflow_breeze/commands/ci_commands.py

```diff
@@ -323,6 +323,8 @@ def is_canary_run(self) -> str:
             and (self.ref_name == "main" or TEST_BRANCH_MATCHER.match(self.ref_name))
         ):
             return "true"
+        if "canary" in self.pull_request_labels and self.head_repo == "apache/airflow":
```

Comment:
I wonder if the second check is really necessary; why not allow Canary from any committer fork?

Reply:
Because they will not work. Canary runs for example push cache to registry and push latest version of docs built to AWS, both of which require "write" tokens, and can only be exacuted in the context of "apache/airflow" repo.

Reply:
See https://github.com/apache/airflow/blob/main/dev/breeze/doc/ci/05_workflows.md#tests-workflow - for the points where have `Yes` only for canary build and  the `(11)` note for docs building.

Reply:
:D

Reply:
Ah okay, I see.

Reply:
Also "constraints" are pushed only in the canary build (and this can't be done from committer fork-run build).